### PR TITLE
Export Curve objects as Godot Path nodes.

### LINKED ISSUE #330: Are there any plans on adding blending Curve import (as Path)?
Are there any plans on adding blender Curve import (as Path)?

The use is to add AI paths in Blender, so to not do hard work in 
(not so handy) Godot editor.

### PR DESCRIPTION
**Pull request by rcorre**
Fixes #330.
Relates to godotengine/godot-proposals#527.

Previously, curves in blender were exported to meshes in Godot.

Now, a bezier curve created in blender will be exported as a Path node
with a Curve3D resource matching the blender data. The curve will have a
child MeshInstance that renders the bevel.

Only bezier curves are exported to Godot Path nodes. Other curves, such
as Nurbs, export only as meshes as they did previously.

### CODE REVIEW

**Comment by rcorre** [comment]
Test scene: Test Scene

**Comment by rcorre** [comment]
So I don't forget, we should update https://docs.godotengine.org/en/3.2/getting_started/workflow/assets/escn_exporter/ if/when this merges.

**Comment by rcorre** [comment]
> RuntimeError: Error: File written by newer Blender binary (290.0), expect loss of data! Looks like I need an older blender to set up the reference files. I'll try to get that set up when I can

**Comment by Jason0214** [comment]
Thanks! 
If it is troublesome to have several versions of Blender, I think we might be able to use the travis CI (or github action) to generate and commit the diff after we verified the change locally.  It should be possible and very helpful for future devs.

**Comment by rcorre** [comment]
That, or maybe a local `Dockerfile`. Is travis docker-based?

**Comment by Calinou** [comment]
> Is travis docker-based?

No, it only offers VMs instead of containers since 2019.

File: io_scene_godot/converters/simple_nodes.py [hunk 1]
```diff
@@ -165,3 +168,68 @@ def export_light_node(escn_file, export_settings, node, parent_gd_node):
...
+    points = Array("PoolVector3Array(")
+    tilts = Array("PoolRealArray(")
+    axis_correct = mathutils.Matrix((
+        (1, 0, 0),  # X in blender is  X in Godot
+        (0, 0, -1),  # Y in blender is -Z in Godot
+        (0, 1, 0),  # Z in blender is  Y in Godot
+    )).normalized()
+
+    src_points = spline.bezier_points
+    if spline.use_cyclic_u:
```
Comment (Jason0214) [inline io_scene_godot/converters/simple_nodes.py:201]: To be robust, might be good to add a check to ensure `src_points` is not empty.

File: io_scene_godot/converters/simple_nodes.py [hunk 2]
```diff
@@ -165,3 +168,68 @@ def export_light_node(escn_file, export_settings, node, parent_gd_node):
...
+
+
+def export_curve_node(escn_file, export_settings, node, parent_gd_node):
+    """Export a curve to a Path node, with a child mesh."""
+    splines = node.data.splines
+    if len(splines) == 0:
+        logging.warning("Curve %s has no splines", node.name)
+        return None
+
+    spline = splines[0]
```
Comment (Jason0214) [inline io_scene_godot/converters/simple_nodes.py:180]: Is it possible that we can export all the spline in the `node.data.splines` and only assign the active one to the Curve node? How do you think? (Btw you can get the active spline by calling `node.data.splines.active`)

**Review by Jason0214 — APPROVED** [review]
Thank you very much adding the curve support!

**Comment by Jason0214** [comment]
> If it is troublesome to have several versions of Blender, I think we might be able to use the travis CI (or github action) to generate and commit the diff after we verified the change locally. It should be possible and very helpful for future devs.

I found it is possible to generate a scene diff patch from the Travis log, I did it in https://github.com/godotengine/godot-blender-exporter/pull/384. I found the `git apply` sometimes fail searching for a very long line, not sure if it's an issue with my local git or a general issue. Anyway, if you are interested you can try that PR out : )

File: io_scene_godot/converters/simple_nodes.py (hunk 1 above)
Comment (rcorre) [reply io_scene_godot/converters/simple_nodes.py:201]: splines with no points are not included in `splines` at all. That is, if you have 1 spline with no points, `splines` actually has `len == 0`.

Comment (Jason0214) [reply io_scene_godot/converters/simple_nodes.py:201]: Got it. Thanks for checking.

**Review by Jason0214 — APPROVED** [review]
It's blocked by a mismatch between the Blender version of test scene (Blender 2.9) and the Blender version that CI runs.

How about removing the test scene temporarily from this change to get it submitted first?
And then bump the Blender version in CI and handle the out-dated test scenes in another change. I can not think of any better ways to do it  : (

**Pull request merged**